tracks/kotlin/exercises/rotational-cipher/mentoring.md

```diff
@@ -0,0 +1,42 @@
+### Reasonable Solutions
+
+```kotlin
+class RotationalCipher(private val key: Int) {
+    init {
+        require(key in 0..26) {
+            "Key should be integer between 0 and 26"
+        }
+    }
+
+    fun encode(input: String): String =
+            input.map(shiftBy(key)).joinToString("")
+
+    // Do not allow private members to access object state.
+    // This allows for easier refactoring to another object
+    // because their state is self contained.
+    private fun shiftBy(key: Int): (Char) -> Char = {
+        if (shouldNotShift(it)) it
+        else doShift(key, it)
+    }
+
+    private fun doShift(key: Int, c: Char): Char {
```

Comment:
Use descriptive names such as `char` or `character`.

Reply:
This seems like another place where we disagree. Would you accept this pull request without it?

I actually prefer the shorter names in this case because the type signature tells me what this thing is...making a longer name doesn't add anything to that understanding but just adds more characters.

Reply:
If this method were longer I would definitely want something more descriptive...I guess this comes from my days of writing Go. :-)

Reply:
This isn't code golf and the Kotlin style guide specifically says to use descriptive names.

Further down the type annotation is not there unless you're in an IDE. I had to scroll up to figure out what c was. I don't want me tors suggesting using non descriptive single character arguments, even though they're not supposed to comment on style in the first place :)

Don't be afraid of characters! :D

Reply:
Good point on this being for mentors and a way to create consistency...I'll make the change. :-)